x/disorder/stream_dialer.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package disorder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/x/sockopt"
+)
+
+type disorderDialer struct {
+	dialer          transport.StreamDialer
+	disorderPacketN int
+}
+
+var _ transport.StreamDialer = (*disorderDialer)(nil)
+
+// NewStreamDialer creates a [transport.StreamDialer]
+// It work almost the same as the other split dialer, however, it also manipulates socket TTL:
+// * Before sending the first prefixBytes TTL is set to 1
+// * This packet is dropped somewhere in the network and never reaches the server
+// * TTL is restored
+// * The next part of data is sent normally
+// * Server notices the lost fragment and requests re-transmission
+// Currently this only works with Linux kernel (for Windows/Mac a different implementation is required)
```

Comment:
Why do you say it needs a different implementation on macOS and Windows?

Reply:
This is from the original PR. Fixed.